coffee/base.py

```diff
@@ -967,3 +1013,26 @@ def __eq__(self, other):
 DEC = Access("DEC")
 IMUL = Access("IMUL")
 IDIV = Access("IDIV")
+Access._modes = [READ, WRITE, RW, INC, DEC, IMUL, IDIV]
+
+
+# Scope of a declaration ##
+
+class Scope(object):
+
+    """An ``EXTERNAL`` scope means the /Decl/ is an argument of a kernel (i.e.,
+    when it appears in the list of declarations of a /FunDecl/ object). Otherwise,
+    a ``LOCAL`` scope indicates the /Decl/ is within the body of a kernel."""
+
+    def __init__(self, scope):
+        self._scope = scope
+
+    def __eq__(self, other):
+        return self._scope == other._scope
```

Comment:
He's saying that if you only use the pre-constructed objects, then object identity is more than satisfactory for comparison.